src/adaptors/http2/http2_adaptor.c

```diff
@@ -2496,10 +2657,21 @@ static void handle_connection_event(pn_event_t *e, qd_server_t *qd_server, void
     }
     case PN_RAW_CONNECTION_WAKE: {
         qd_log(log, QD_LOG_TRACE, "[C%"PRIu64"] PN_RAW_CONNECTION_WAKE Wake-up", conn->conn_id);
+        if (IS_ATOMIC_FLAG_SET(&conn->q2_restart)) {
```

Comment:
This needs to be an atomic set and clear, otherwise you'll miss transitions.
Use CLEAR_ATOMIC_FLAG and check the return value for true.

Reply:
Added CLEAR_ATOMIC_FLAG. Check the return value of CLEAR_ATOMIC_FLAG ?

Reply:
Precisely - SET and CLEAR operations must return the *previous* value, so if (CLEAR_ATOMIC_FLAG()) is true you know it went from True->False ... atomically.

We don't have a conditional test-and-set atomic operation.. yet.